NN-notebook-v7/notebook-v7.md

```diff
@@ -0,0 +1,118 @@
+
+This document proposes that the next major release of the [Jupyter Notebook](https://jupyter-notebook.readthedocs.io) application, version 7, will be based on the JupyterLab codebase, but will provide an equivalent user experience to the current (version 6) application.  Jupyter Notebook version 7 ("Notebook v7" henceforth) will achieve this by using the current version of [RetroLab](https://github.com/jupyterlab/retrolab). The following three key ideas are central to this planned transition:
+
+**The Notebook as a document-centric user experience:** The Jupyter Notebook application offers a _document-centric user experience_. That is, in the Notebook application, the landing page that contains a file manager, running tools tab, and a few optional extras, is a launching point into opening standalone, individual documents. This document-centric experience is important for many users, and that is the first key point this proposal aims to preserve. Notebook v7 will be based on a different JavaScript implementation than v6, but _it will preserve the document-centric experience, where each individual notebook opens in a separate browser tab and the visible tools and menus are focused on the open document_.
+
+**Extensions critical to the Notebook user community:** Furthermore, for many users, certain widely used extensions are critical to their workflow. While we can not port every Notebook v6 extension ourselves (a list of widely-used community extensions is [here](https://github.com/ipython-contrib/jupyter_contrib_nbextensions)), we will identify some critical extensions (listed below) for which, on day 1 of the Notebook v7 release, should work with similar functionality that users currently enjoy. We will engage with the developers of those extensions to identify a concrete technical plan and resources to implement this transition. We will also engage with the broader extension-authoring community to provide resources to ease the transition of other extensions.
+
+**The body of existing content relying on the Notebook application experience:** Finally, the Notebook v6 community depends on widely-used educational content about the core Notebook application (such as tutorials, textbooks, videos), and these _will not be invalidated_ by the transition. Educators who have such materials online will not need to update their content. While we can not expect a pixel-perfect match between versions 6 and 7 of the Notebook, we will make every effort to ensure that the gap between the visual and operational experience (menu items, available tools and actions, etc.) is small enough that any user can reasonably bridge it without assistance. A user's experience applying a Notebook v6 tutorial to Notebook v7 should be no different than that of reading an online tutorial based on a previous version of Windows or MacOS, where the visuals or item locations may have changed somewhat in the user's current installation.
```

Comment:
I think this paragraph, as written, is ambiguous. I think the ideas of "there will be bridgable gaps" and "like is like updates to Windows or MacOS version updates" are a good starting point, but in reality, I think the gaps - even though bridgable by users - will require updating educational materials depending on what they cover. Sure, generic things will be the same, but there are plenty of old JupyterLab tutorials we have given that we would never recommend to someone now because the changes are so significant. 

I believe what we are saying is more something like this: content that uses notebook <=v6 to teach primarily about other things (matplotlib, data science, ML, physics, etc.) will not need to be updated, however content about Jupyter itself may require some updates. The scenarios we brought up when we talked about this initially (professors who have years worth of notebooks that teach an academic topic) is consistent with this.

Reply:
> I think this paragraph, as written, is ambiguous. I think the ideas of "there will be bridgable gaps" and "like is like updates to Windows or MacOS version updates" are a good starting point, but in reality, I think the gaps - even though bridgable by users - will require updating educational materials depending on what they cover. Sure, generic things will be the same, but there are plenty of old JupyterLab tutorials we have given that we would never recommend to someone now because the changes are so significant.

I think that the main meaning of the paragraph is that tutorials which are not specifically *about* jupyter or extending Jupyter will just continue working and look the same (such as a scikit-learn tutorial with screenshots to notebooks).

Reply:
```suggestion
**The body of existing content relying on the Notebook application experience:** Finally, the Notebook v6 community depends on widely-used educational content using the core Notebook application (such as tutorials, textbooks, videos), and these _will not be invalidated_ by the transition. Educators who have such materials online will not need to update their content. While we can not expect a pixel-perfect match between versions 6 and 7 of the Notebook, we will make every effort to ensure that the gap between the visual and operational experience (menu items, available tools and actions, etc.) is small enough that any user can reasonably bridge it without assistance.
```